src/app/component/dashboard/vaucher/vaucher.component.ts

```diff
@@ -262,7 +288,14 @@ export class VaucherComponent implements OnInit {
     }
     this.data.date = this.dateConst.toString();
     this.data.date_redeemed = this.dateredeemedConst.toString();
-    this.service.createVaucher(this.data).subscribe(data => {
+    this.data["language"] = this.packLanguage.getLanguageForMailingVaucher();
+    this.data["toSendEmail"] = this.toSendEmail;
+
+    if (this.toSendSms) {
```

Comment:
I would say it is better to first create voucher and then to send email. Easily can happen that mail is sent and voucher not created. Please check this.